File: modules/core-api/src/main/java/com/enonic/wem/api/content/ContentPublishedEvent.java
package com.enonic.wem.api.content;

import com.google.common.base.Objects;

import com.enonic.wem.api.event.Event;

public final class ContentPublishedEvent
    implements Event
{
    private final ContentId contentId;

    public ContentPublishedEvent( final ContentId contentId )
    {
        this.contentId = contentId;
    }

    public ContentId getContentId()
    {
        return contentId;
    }

    @Override
    public String toString()
    {
        return Objects.toStringHelper( this ).
            add( "contentId", this.contentId ).
            omitNullValues().
            toString();
    }

}

File: modules/admin/admin-ui/src/main/resources/web/admin/common/js/content/ContentPublishedEvent.ts
module api.content {

    export class ContentPublishedEvent extends api.event.Event {

        private contentId: api.content.ContentId;

        private compareStatus: api.content.CompareStatus;

        constructor(contentId: api.content.ContentId, compareStatus: api.content.CompareStatus) {
            super();
            this.contentId = contentId;
            this.compareStatus = compareStatus;
        }

        public getContentId(): api.content.ContentId {
            return this.contentId;
        }

        public getCompareStatus(): api.content.CompareStatus {
            return this.compareStatus;
        }

        static on(handler: (event: ContentPublishedEvent) => void) {
            api.event.Event.bind(api.ClassHelper.getFullName(this), handler);
        }

        static un(handler?: (event: ContentPublishedEvent) => void) {
            api.event.Event.unbind(api.ClassHelper.getFullName(this), handler);
        }

    }
}